From 9970248dfbc5f394152ee1d15e3106ae28c11110 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Sat, 20 Feb 2016 00:56:47 +0200
Subject: [PATCH 25/55] SystemUI: Don't vibrate on touchscreen camera gesture

* This is usually handled in device-specific keyhandler,
  where vibration is optional and has the same duration
  in all gestures

Change-Id: I879671c9d6f785137d63b8b70de35f5d5020748d
---
 .../src/com/android/systemui/statusbar/phone/StatusBar.java   | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
index b4353dc9b02..ac7b36bcc24 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
@@ -4862,7 +4862,9 @@ public class StatusBar extends SystemUI implements DemoMode, TunerService.Tunabl
             pm.wakeUp(SystemClock.uptimeMillis(), "com.android.systemui:CAMERA_GESTURE");
             mStatusBarKeyguardViewManager.notifyDeviceWakeUpRequested();
         }
-        vibrateForCameraGesture();
+        if (source != StatusBarManager.CAMERA_LAUNCH_SOURCE_SCREEN_GESTURE) {
+            vibrateForCameraGesture();
+        }
         if (!mStatusBarKeyguardViewManager.isShowing()) {
             startActivityDismissingKeyguard(KeyguardBottomAreaView.INSECURE_CAMERA_INTENT,
                     false /* onlyProvisioned */, true /* dismissShade */,
-- 
2.17.1

